test/intest1.F90

```diff
@@ -100,6 +101,9 @@ program intest1
   call readlc(11, softvstg, 'SOFTV')
   IF (softvstg(5:12) .ne. '5.8.5.10') stop 17
 
+  call getlens(ibfmg,3,l0,l1,l2,l3,l4,l5)
```

Comment:
Can you give me a one-line comment as to what is happening?

Reply:
`getlens` returns the lengths of BUFR sections in the `ibmfg` bufr message (up to and including section 3, in this case). Because `setim8b` has been invoked above, the `IF(IM8B)` clause is triggered so it returns 8-byte integers for the section lengths.

Reply:
Yes, but add that explanation to the code as a comment. ;-)

Reply:
Any chance you could modify the test to check all the way through Section 5 so that **all** of the lines of getlens get tested, and not just the ones that do Sections 0 through 3? ;-)

Also, why do you have variable l6 declared?  Unless I'm missing something, it looks like it's unused.

Finally, I would suggest to take the "(returns 8-byte ints for KIND_8)" out of the comment, because that's no different than for any other subroutine call as well.  In other words, just "Get and check lengths of Sections 0-3 (or 5 ;-)" should be enough.